lib/alert.js

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2017 ODK Central Developers
+See the NOTICE file at the top-level directory of this distribution and at
+https://github.com/opendatakit/central-frontend/blob/master/NOTICE.
+
+This file is part of ODK Central. It is subject to the license terms in
+the LICENSE file found in the top-level directory of this distribution and at
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+*/
+
+/*
+A component that is the parent of an Alert component must define a property in
+data() named 'alert', containing the alert's underlying data. The property must
+be an object that has the following properties:
+
+  - state. true if the alert is visible and false if not.
+  - type. The alert's "contextual" type: 'success', 'info', 'warning', or
+    'danger'.
+  - message. The message to display.
+  - at. The time at which the data object was created.
+*/
+
+// Returns a data object for a new visible alert that is not blank.
+export const newAlert = (type, message) => ({
+  state: true,
+  type,
+  message,
+  at: new Date()
+});
+
+// Returns a data object for a new blank alert.
+export const blankAlert = (type = 'danger') => ({
+  state: false,
+  type,
+  message: null,
+  at: new Date()
+});
+
+// Convenience class used to make it easier to modify a component's alert
```

Comment:
in what way? this comment doesn't really make sense on its own without looking for usages.

Reply:
Good point, I'll update this comment!